specs/challenge.md

```diff
@@ -110,212 +148,325 @@ You can notice that in each turn, the first element of the segments must be same
-   until time `t`".
-
-2. At `t + 3 hours 30 minutes`, david initiates a challenge to the output at 5400.
-
-   | Name  | Output | Challenge    | Bond | Lock                     |
-   |-------|--------|--------------|------|--------------------------|
-   | alice | O_1800 | N            | 100  | L_{t + 7 days}           |
-   | bob   | O_3600 | N            | 100  | L_{t + 7 days + 1 hours} |
-   | bob   | O_5400 | Y (by david) | 200  | L_{t + 7 days + 2 hours} |
-   | carol | O_7200 | N            | 100  | L_{t + 7 days + 3 hours} |
-
-3. At `t + 4 hours`, emma submits a output at 9000.
-
-   | Name  | Output | Challenge    | Bond | Lock                     |
-   |-------|--------|--------------|------|--------------------------|
-   | alice | O_1800 | N            | 100  | L_{t + 7 days}           |
-   | bob   | O_3600 | N            | 100  | L_{t + 7 days + 1 hours} |
-   | bob   | O_5400 | Y (by david) | 200  | L_{t + 7 days + 2 hours} |
-   | carol | O_7200 | N            | 100  | L_{t + 7 days + 3 hours} |
-   | emma  | O_9000 | N            | 100  | L_{t + 7 days + 4 hours} |
-
-4. If the challenger wins:
-
-   | Name  | Output | Challenge | Bond | Lock                     |
-   |-------|--------|-----------|------|--------------------------|
-   | alice | O_1800 | N         | 100  | L_{t + 7 days}           |
-   | bob   | O_3600 | N         | 100  | L_{t + 7 days + 1 hours} |
-   | david | O_5400 | N         | 200  | L_{t + 7 days + 2 hours} |
-   | carol | O_7200 | N         | 100  | L_{t + 7 days + 3 hours} |
-   | emma  | O_9000 | N         | 100  | L_{t + 7 days + 4 hours} |
-
-5. Otherwise:
-
-   | Name  | Output | Challenge | Bond | Lock                     |
-   |-------|--------|-----------|------|--------------------------|
-   | alice | O_1800 | N         | 100  | L_{t + 7 days}           |
-   | bob   | O_3600 | N         | 100  | L_{t + 7 days + 1 hours} |
-   | bob   | O_5400 | N         | 200  | L_{t + 7 days + 2 hours} |
-   | carol | O_7200 | N         | 100  | L_{t + 7 days + 3 hours} |
-   | emma  | O_9000 | N         | 100  | L_{t + 7 days + 4 hours} |
-
-## Public Input Verification
-
-Since Colosseum verifies public input along with [zkevm-proof](./zkevm-prover.md#zkevm-proof), challengers should
+Since Colosseum verifies public input along with [zkEVM-proof](./zkevm-prover.md#zkevm-proof), challengers should
 calculate as below and enclose the public input to the `proveFault` transaction.
 
 ```ts
-  import { DataOptions, hexlify } from '@ethersproject/bytes';
-  import { Wallet, constants } from 'ethers';
-  import { keccak256 } from 'ethers/lib/utils';
+import { DataOptions, hexlify } from '@ethersproject/bytes';
+import { Wallet, constants } from 'ethers';
+import { keccak256 } from 'ethers/lib/utils';
 
-  function strip0x(str: string): string {
+function strip0x(str: string): string {
     if (str.startsWith('0x')) {
-      return str.slice(2);
+        return str.slice(2);
```

Comment:
We're using 2 spaces as tab width, so it's better to delete the indent you added.